types/test.ts

```diff
@@ -4,6 +4,12 @@ import * as Knex from 'knex';
 // import Knex from 'knex'
 // when "esModuleInterop": true
 
+declare module 'knex' {
+  interface QueryBuilder {
+    onDuplicateUpdate(...columnNames: string[]): QueryBuilder;
```

Comment:
Are any assertions possible on existence of this method?

Reply:
I've improved it & added a usage. 
:]